File: Backend/AICareerHub.API/AICareerHub.API/Models/ResumeExperience.cs
namespace AICareerHub.API.Models
{
    public class ResumeExperience
    {
        public Guid Id { get; set; }

        public Guid ResumeId { get; set; }

        public string Company { get; set; } = string.Empty;

        public string JobTitle { get; set; } = string.Empty;

        public DateOnly StartDate { get; set; }

        public DateOnly? EndDate { get; set; }

        public bool IsCurrent { get; set; }

        public string? Description { get; set; }

        public Resume Resume { get; set; } = null!;
    }
}

File: Frontend/aicareerhub-ui/src/app/core/models/resume-experience.ts
export interface ResumeExperience {
  id: string;
  company: string;
  jobTitle: string;
  startDate: string;
  endDate?: string | null;
  isCurrent: boolean;
  description?: string | null;
}

export interface ResumeExperienceRequest {
  company: string;
  jobTitle: string;
  startDate: string;
  endDate?: string | null;
  isCurrent: boolean;
  description?: string | null;
}